mailpoet/lib/Automation/Integrations/Core/Filters/IntegerFilter.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\Core\Filters;
+
+use MailPoet\Automation\Engine\Data\Field;
+use MailPoet\Automation\Engine\Data\Filter as FilterData;
+use MailPoet\Validator\Builder;
+use MailPoet\Validator\Schema\ObjectSchema;
+
+class IntegerFilter extends NumberFilter {
+  public function getFieldType(): string {
+    return Field::TYPE_INTEGER;
+  }
+
+  public function getArgsSchema(): ObjectSchema {
+    return Builder::object([
+      'value' => Builder::oneOf([
+        Builder::integer()->required(),
+        Builder::array(Builder::integer())->minItems(2)->maxItems(2)->required(),
+      ]),
+    ]);
+  }
+
+  public function matches(FilterData $data, $value): bool {
+    $matches = parent::matches($data, $value);
+    if (!$matches) {
+      return false;
+    }
+
+    if (isset($value) && !$this->isWholeNumber($value)) {
+      return false;
+    }
+
+    $filterValue = $data->getArgs()['value'] ?? null;
+    if (is_array($filterValue)) {
```

Comment:
If I understand this correctly `$filterValue` can't be an array as `parent::matches()` would return `false` in the case `$filterValue` is not a number.

Reply:
Ah! `matchesBetween()` returns early in the `parent`. Ok, my fault.